meja/envi.ml

```diff
@@ -11,6 +11,7 @@ type error =
   | Wrong_number_args of string * int * int
   | Expected_type_var of type_expr
   | Lident_unhandled of string * Longident.t
+  | Constraints_not_satisfied of type_expr * type_decl
```

Comment:
what does this error mean exactly?

Reply:
The type constraint on a constructor of a variant type doesn't match the variant type that owns it. So like
```ocaml
type _ t = A : bool;;
```

The wording here is based off the OCaml error message in the same situation.

Reply:
Ah got it